snapcraft/internal/parser.py

```diff
@@ -140,6 +144,73 @@ def _process_subparts(project_part, subparts, parts, origin, maintainer,
     return parts_list, after_parts
 
 
+def _process_entry(data):
+    key = data.get('project-part')
+    parts_list = {}
+    # Store all the parts listed in 'after' for each included part so that
+    # we can check later that we aren't missing any parts.
+    # XXX: What do we do about 'after' parts that should be looked for in
+    # the wiki?  They should be in the master parts list.
+    after_parts = set()
+
+    logger.debug('Processing part {!r}'.format(key))
+
+    try:
+        origin = data['origin']
+        origin_type = data.get('origin-type')
+        project_part = data['project-part']
+        subparts = data.get('parts', [])
+        maintainer = data['maintainer']
+        description = data['description']
+    except KeyError as e:
+        raise InvalidEntry("Missing key in wiki entry: {}".format(e))
```

Comment:
That's okay, it would probably break the case where @elopio prefers `"here is my 'quoted' thing"` to `'here is my \'quoted\' thing` anyway :stuck_out_tongue: